modules/gapi/src/backends/oak/goakbackend.cpp

```diff
@@ -0,0 +1,616 @@
+
+        // 2. Link output nodes to XLinkOut nodes
+        size_t out_counter = 0;
+        for (const auto& d : outs_data)
+        {
+            for (const auto& nh : nodes)
+            {
+                if (m_gm.metadata(nh).contains<cv::gimpl::Op>())
+                {
+                    for (const auto& outdata : nh.get()->outNodes()) {
+                        auto rc = m_gm.metadata(outdata).get<cv::gimpl::Data>().rc;
+                        if (rc == d.rc)
+                        {
+                            GAPI_Assert(m_oak_nodes.find(nh) != m_oak_nodes.end());
+                            GAPI_Assert(m_oak_node_outputs[nh].size() == 1);
+                            GAPI_Assert(out_counter < m_xlink_outputs.size());
+                            m_oak_node_outputs[nh][0]->link(m_xlink_outputs[out_counter++]->input);
+
+                            if (m_processed_nodes.find(nh) == m_processed_nodes.end()) {
+                                LinkToParentHelper(nh, nodes);
+                                m_processed_nodes.insert(nh);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // 3. Link internal nodes to their parents
+        for (const auto& nh : nodes) {
+            if (m_gm.metadata(nh).get<NodeType>().t == NodeType::OP) {
+                GAPI_Assert(m_oak_nodes.find(nh) != m_oak_nodes.end());
+                if (m_processed_nodes.find(nh) == m_processed_nodes.end()) {
+                    LinkToParentHelper(nh, nodes);
+                    m_processed_nodes.insert(nh);
+                }
+            }
+        }
+
+        m_device = std::unique_ptr<dai::Device>(new dai::Device(*m_pipeline));
+
+        // Prepare all output queues
+        for (size_t i = 0; i < outs_data.size(); ++i) {
+            // FIXME: add queue parameters
+            m_out_queues.push_back(m_device->getOutputQueue(m_out_queue_names[i], 30, true));
+        }
+    }
+
+void cv::gimpl::GOAKExecutable::handleNewStream() {
+    // do nothing
+}
+
+void cv::gimpl::GOAKExecutable::handleStopStream() {
+    // do nothing
+}
+
+void cv::gimpl::GOAKExecutable::run(GIslandExecutable::IInput  &in,
+                                    GIslandExecutable::IOutput &out) {
+    const auto in_msg = in.get();
+
+    if (cv::util::holds_alternative<cv::gimpl::EndOfStream>(in_msg)) {
+        out.post(cv::gimpl::EndOfStream{});
+        return;
+    }
+
+    for (size_t i = 0; i < m_in_queues.size(); ++i) {
+        auto q = m_device->getInputQueue(m_in_queues[i].first);
+        q->send(m_in_queues[i].second);
+    }
+
+    for (size_t i = 0; i < m_out_queues.size(); ++i) {
+        auto q = m_out_queues[i];
+        auto oak_frame = q->get<dai::ImgFrame>();
+
+        auto out_arg = out.get(i);
+
+        if (util::holds_alternative<cv::detail::VectorRef>(out_arg)) {
+            cv::util::get<cv::detail::VectorRef>(out_arg).wref<uint8_t>() = oak_frame->getData();
```

Comment:
It shouldn't work like this, please check the Meta type of the data object (shape or smt) rather than make decisions on types stored in the container.

Reply:
Done